[BD-38] fix: hide discussion xblock for openedx provider

navinkarkera opened the pull request:
<!--

🌰🌰
🌰🌰🌰🌰         🌰 Note: the Nutmeg master branch has been created.  Please consider whether your change
    🌰🌰🌰🌰     should also be applied to Nutmeg. If so, make another pull request against the
🌰🌰🌰🌰         open-release/nutmeg.master branch, or ping @nedbat for help or questions.
🌰🌰

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply. You may link to information rather than copy it.
More details about the template are at https://github.com/edx/open-edx-proposals/pull/180
(link will be updated when that document merges)
-->

## Description

Open edX now support multiple discussion providers. One of them is the "legacy" provider and another is the newer "openedx" provider. The Discussion XBlock is only supported for the older legacy provider.

~This MR hides the xblock if provider is set to `openedx`.~
This MR hides the xblock if provider is not set to `legacy` as well as adds a message in studio.

## Testing instructions

1. Setup course with discussion xblock or use demo course which has a discussion xblock added in chapter 3: Be social.
2. First make sure that you can see the discussion xblock
3. Goto http://localhost:18000/admin/discussions/discussionsconfiguration/ and select the course.
4. Change `Discussion provider` to "openedx"
5. Reload the course chapter which contains discussion xblock. It should not visible. 
6. Reload studio and verify that a message appears in discussion block stating that the block will not be visible.

## Other information

`Private-ref:` [BB-6509](https://tasks.opencraft.com/browse/BB-6509)

navinkarkera committed:
fix: hide discussion xblock for openedx provider

openedx-webhooks commented:
Thanks for the pull request, @navinkarkera!


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

navinkarkera committed:
fix: test cases

natabene commented:
@navinkarkera Thank you for your contribution. We will try to review is in the coming weeks.

navinkarkera commented:
Current implementation is hiding discussion xblock when provider is `openedx`, but will it make more sense to hide discussion xblock for all providers except `legacy` as it is the only one supported if I am not wrong.

navinkarkera committed:
refactor: hide xblock is provider not legacy and add message in studio

navinkarkera committed:
fix: lint issue

navinkarkera commented:
Updated the PR to display discussion xblock only if provider is set to legacy and added a message in studio if the block will not be visible.

cc: @xitij2000

xitij2000 commented on lms/templates/discussion/_discussion_inline_studio.html:
```diff
@@ -7,6 +7,9 @@
             ${_("Discussion ID: {discussion_id}").format(discussion_id=discussion_id)}
+            % if not is_visible:
+            <br /><b>${_('Due to the discussion provider not being set to "legacy", this block will not be visible.')}</b>
+            % endif
```
I think we shouldn't put this technical information here. 

Instead, this should say something along the lines of:
"The discussion block is disabled for this course since it isn't using a compatible discussion provider. Read more here"

navinkarkera replied:
@xitij2000 Makes sense. Do we have any links for `Read more here`?

navinkarkera replied:
@xitij2000 Updated the message but did not include any links.

xitij2000 replied:
Not yet. I think we can get some feedback first.

xitij2000 reviewed (approved):
:+1:

- [x] I tested this: tested on devstack
- [x] I read through the code
- [na] I checked for accessibility issues
- [na] Includes documentation

navinkarkera committed:
refactor: update discussion block studio msg

asadazam93 commented:
@navinkarkera can you please rebase this?

navinkarkera commented:
@asadazam93 Done.

asadazam93 commented:
@navinkarkera Some unit tests are failing. Can you please check?

navinkarkera committed:
fix: test discussion xblock query number

openedx-webhooks commented:
@navinkarkera 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/30845&entry.1484655318=https://github.com/navinkarkera&entry.752974735=2022-08-11+10:21&entry.1917517419=2022-09-02+09:38&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
